Sep sales and acct pages

mmdotz opened the pull request.

mmdotz committed:
non-subscribers should keep option to subscribe in nav bar

mmdotz committed:
correct nav bar views for current_users subscribed, admin, signed in

mmdotz committed:
add conditional for log-in button on locked out post show

mmdotz committed:
create partial for sales, sep from account

mmdotz committed:
change partial to account sales page for route

jeffreybiles commented on app/views/posts/show.html.erb:
```diff
@@ -35,7 +35,9 @@
           <% else %>
             <div class="sales-box">
               <h2>You can view pro screencasts by subscribing</h2>
+              <% if current_user && current_user.subscribed %>
```
This conditional logic doesn't seem right.  We want the button to show if it's an anonymous user.

mmdotz replied:
fixed

jeffreybiles commented on app/views/layouts/application.html.erb:
```diff
@@ -53,12 +53,16 @@
+              <li><a href="/account">Subscribe to Pro</a></li>
+              <% else %>
+                <li><%= link_to current_user.email, account_path %></li>
+              <% end %>
```
indentation is confusing.  unless/else/end should line up, as should the two `<li>`'s

mmdotz replied:
fixed

jeffreybiles commented on app/views/layouts/application.html.erb:
```diff
@@ -53,12 +53,16 @@
         </form> -->
         <ul class="nav navbar-nav navbar-right">
           <% if current_user %>
+            <% unless current_user.subscribed %>
```
Using unless/else is an anti-pattern.  Use if/else and reverse the order, unless there's a really good reason not to.

mmdotz replied:
fixed

jeffreybiles commented on app/views/layouts/application.html.erb:
```diff
@@ -53,12 +53,16 @@
+            <% unless current_user.subscribed %>
+              <li><a href="/account">Subscribe to Pro</a></li>
+              <% else %>
+                <li><%= link_to current_user.email, account_path %></li>
```
We want unsubscribed people to still be able to access their account.

mmdotz replied:
From your email: 1) So what I currently call the “sales page” is what anonymous users and unsubscribed users see when viewing the account page.

Are you asking for a separate account page for "unsubscribed but signed in" users instead? What markup is used for an unsubscribed user account page? The current account page code wouldn't be valid.

jeffreybiles commented on app/views/accounts/sales.html.erb:
```diff
@@ -0,0 +1,23 @@
+
+  <ul>
+      <li>New up-to-date videos every week.</li>
+      <li>Over 7 hours of fast-paced content.</li>
```
While you're updating stuff, could you change this 7 to 10? I've made a lot of screencasts since I wrote this :)

mmdotz replied:
fixed

jeffreybiles commented on app/views/accounts/show.html.erb:
```diff
@@ -20,39 +20,21 @@
-    <!-- Another subscribe button -->
-
-    <% content_for :title, "Learn like a Pro- sign up for an EmberScreencasts Pro Subscription." %>
+  <%= controller.redirect_to sales_path %>
```
This is an unusual construct- typically I would expect to see this redirect in the controller.  Have conventions changed around this recently?

mmdotz replied:
No, you're right this isn't typical. I substituted one line of code in the view for before_actions and custom methods in the controller. I can change that if you prefer.

> On Jan 27, 2016, at 4:45 PM, Jeffrey Biles notifications@github.com wrote:
> 
> In app/views/accounts/show.html.erb https://github.com/jeffreybiles/simpler-ember-sparks-website/pull/6#discussion_r51063097:
> 
> > -        <li>Over 7 hours of fast-paced content.</li>
> > -        <li>No commitment- unsubscribing is instant and easy.</li>
> > 
> > ## \-    </ul>
> > -    <h2>Save time.  Learn more.  Be better.</h2>
> > -    <%= render 'shared/subscribe' %>
> > -    <br>
> > -    <a href='https://www.expeditedssl.com/simple-ssl-scanner/scan?target_domain=www.emberscreencasts.com' target='_blank'><img src='https://www.expeditedssl.com/ssl-secure-badge.png'  alt='Expedited SSL Scanner'></a>
> > -    <br>
> > 
> > ## \-    <p>Buying for your organization?  You can add more users to your account after signing up.</p>
> > -    <!-- Lots of testimonials -->
> > 
> > ## \-    <!-- Another subscribe button -->
> > -    <% content_for :title, "Learn like a Pro- sign up for an EmberScreencasts Pro Subscription." %>
> > -  <%= controller.redirect_to sales_path %>
> >   This is an unusual construct- typically I would expect to see this redirect in the controller. Have conventions changed around this recently?
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/jeffreybiles/simpler-ember-sparks-website/pull/6/files#r51063097.

jeffreybiles replied:
Yea let's change it- always better to do it the happy path way when we can.

mmdotz replied:
fixed

> On Jan 27, 2016, at 9:58 PM, Jeffrey Biles <notifications@github.com <mailto:notifications@github.com>> wrote:
> 
> In app/views/accounts/show.html.erb https://github.com/jeffreybiles/simpler-ember-sparks-website/pull/6#discussion_r51077296:
> 
> > -        <li>Over 7 hours of fast-paced content.</li>
> > -        <li>No commitment- unsubscribing is instant and easy.</li>
> > 
> > ## \-    </ul>
> > -    <h2>Save time.  Learn more.  Be better.</h2>
> > -    <%= render 'shared/subscribe' %>
> > -    <br>
> > -    <a href='https://www.expeditedssl.com/simple-ssl-scanner/scan?target_domain=www.emberscreencasts.com' <https://www.expeditedssl.com/simple-ssl-scanner/scan?target_domain=www.emberscreencasts.com'> target='_blank'><img src='https://www.expeditedssl.com/ssl-secure-badge.png' <https://www.expeditedssl.com/ssl-secure-badge.png'>  alt='Expedited SSL Scanner'></a>
> > -    <br>
> > 
> > ## \-    <p>Buying for your organization?  You can add more users to your account after signing up.</p>
> > -    <!-- Lots of testimonials -->
> > 
> > ## \-    <!-- Another subscribe button -->
> > -    <% content_for :title, "Learn like a Pro- sign up for an EmberScreencasts Pro Subscription." %>
> > -  <%= controller.redirect_to sales_path %>
> >   Yea let's change it- always better to do it the happy path way when we can.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/jeffreybiles/simpler-ember-sparks-website/pull/6/files#r51077296.

mmdotz committed:
sales page update hours of footage

mmdotz committed:
correct indentation and conditional antipattern for Subscribe action

mmdotz committed:
anon users should see log in button on locked post show page

jeffreybiles commented on app/views/layouts/application.html.erb:
```diff
@@ -53,12 +53,16 @@
+            <% else %>
+              <li><a href="/account">Subscribe to Pro</a></li>
+            <% end %>
+              <li><%= link_to 'Sign Out', destroy_user_session_path, method: :delete %></li>
```
I would indent this line in with the if/else statement

mmdotz replied:
To confirm, you prefer the html tags within if/else's to be equal indentation? If so, I will change that. Is that convention? I've been taught its more personal preference, and I find it easier to read, but I would rather learn the way seasoned Rails developers expect it to appear. Thanks for the guidance around these topics.

jeffreybiles replied:
So the typical indentation I've seen is, if you have an opening and closing tag, then everything in between gets indented another line.  So the two `li`s within the if/else/end block are indented an extra line in, but the one after (not between the opening if and closing end) does not have the extra indent.

jeffreybiles commented on app/views/layouts/application.html.erb:
```diff
@@ -53,12 +53,16 @@
         <ul class="nav navbar-nav navbar-right">
           <% if current_user %>
+            <% if current_user.subscribed %>
+              <li><%= link_to current_user.email, account_path %></li>
```
I see what you mean about needing a separate page for "signed in but unsubscribed".  Disregard my previous comment and carry on as-is for now.

jeffreybiles commented:
Looks awesome!  After moving that bit of functionality from template to controller it should be good to merge.

You can go ahead and branch from here for the new billing/testing branch, so that you're not waiting on my merge.

mmdotz committed:
redirect unsubscribed users to sales page in controller

jeffreybiles committed:
I want non-pro users to be able to see an account page.  This lets them do that.

Currently their account page is rather empty, but it will eventually grow to something larger.

jeffreybiles committed:
Merge pull request #9 from jeffreybiles/account-page-for-non-pro-members

Account page for non-pro users

jeffreybiles committed:
refactor if statement in show view

jeffreybiles commented on config/routes.rb:
```diff
@@ -12,13 +12,14 @@
 
   get 'set_password/:user_id/:token', to: 'users#set_password'
-  post 'set_password/:user_id/:token', to: 'users#activate_organization_user'
+  post 'set_password/:user_id/:token', to: 'users#açctivate_organization_user'
```
this would have been bad to let through.  Fixing.  Another reason to always review your own changes file by file (in something like gitx) before committing.

jeffreybiles committed:
fixing typo in routes
